pkg/controllers/updaterun/execution.go

```diff
@@ -122,7 +122,7 @@ func (r *Reconciler) executeUpdatingStage(
 		binding := toBeUpdatedBindingsMap[clusterStatus.ClusterName]
 		if !condition.IsConditionStatusTrue(clusterStartedCond, updateRun.Generation) {
 			// The cluster has not started updating yet.
-			if !isBindingSyncedWithClusterStatus(resourceSnapshotName, updateRun, binding, clusterStatus) {
+			if diff := getBindingDiffFromClusterStatus(resourceSnapshotName, updateRun, binding, clusterStatus); diff != "" {
```

Comment:
where is the diff used?  It seems that we called the getBindingDiffFromClusterStatus function again later?

Reply:
Function restored.